test/NuGet.Core.Tests/NuGet.Packaging.Test/DependeciesGroupsForEachTFMRuleTests.cs

```diff
@@ -0,0 +1,242 @@
+                                                        AnalysisResources.DependenciesGroupsForEachTFMHasCompatMatch);
+            byte[] byteArray = Encoding.ASCII.GetBytes(CreateNuspecContent(frameworks));
+            MemoryStream stream = new MemoryStream(byteArray);
+            var issues = rule.Validate(files, stream).ToList();
+
+            // Assert
+            Assert.True(issues.Any(p => p.Code == NuGetLogCode.NU5128));
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5130));
+        }
+
+        [Fact]
+        public void Validate_PackageHasCompatMatchFromNuspecToPackage_ShouldWarnOnce()
+        {
+            var frameworks = new string[]
+            {
+                ".NETFramework4.7.2"
+            };
+            var files = new string[]
+            {
+                "lib/net45/test.dll"
+            };
+            // Act
+            var rule = new DependeciesGroupsForEachTFMRule(AnalysisResources.DependenciesGroupsForEachTFMHasNoExactMatch,
+                                                        AnalysisResources.DependenciesGroupsForEachTFMHasCompatMatch);
+            byte[] byteArray = Encoding.ASCII.GetBytes(CreateNuspecContent(frameworks));
+            MemoryStream stream = new MemoryStream(byteArray);
+            var issues = rule.Validate(files, stream).ToList();
+
+            // Assert
+            Assert.True(issues.Any(p => p.Code == NuGetLogCode.NU5128));
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5130));
+        }
+
+        [Fact]
+        public void Validate_PackageHasNoDepencyNodeAndHasFilesInLib_ShouldNotWarn()
+        {
+            var frameworks = Array.Empty<string>();
+            var files = new string[]
+            {
+                "lib/test.dll"
+            };
+            // Act
+            var rule = new DependeciesGroupsForEachTFMRule(AnalysisResources.DependenciesGroupsForEachTFMHasNoExactMatch,
+                                                        AnalysisResources.DependenciesGroupsForEachTFMHasCompatMatch);
+            byte[] byteArray = Encoding.ASCII.GetBytes(CreateNuspecContent(frameworks));
+            MemoryStream stream = new MemoryStream(byteArray);
+            var issues = rule.Validate(files, stream).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5128));
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5130));
+        }
+
+        public static string CreateNuspecContent(string[] frameworks)
+        {
+            string dependencies = "<dependencies>";
+            foreach(var framework in frameworks)
+            {
+                dependencies = dependencies + "<group targetFramework= \"" + framework + "\" />";
+            }
+            dependencies = dependencies + "</dependencies>";
+            return NuspecContent.Replace("</language>", "</language>" + dependencies);
+        }
+    }
+}
+
+
+    
```

Comment:
formatting no empty lines at end of file (but the last line should end with a new line. if you rebase your changes to the latest dev branch, this will happen automatically).